File: smartdevices-app/smartdevicesapp.model/src/main/java/de/vogler_engineering/smartdevicesapp/model/entities/dto/DeviceInfo.java
/**
 * Copyright (c) Vogler Engineering GmbH. All rights reserved.
 * Licensed under the MIT License. See LICENSE.md in the project root for license information.
 */
package de.vogler_engineering.smartdevicesapp.model.entities.dto;

import java.util.HashMap;
import java.util.Map;

import de.vogler_engineering.smartdevicesapp.model.entities.enums.DeviceFamily;
import de.vogler_engineering.smartdevicesapp.model.entities.enums.DeviceType;
import lombok.Data;

/**
 * Created by vh on 14.02.2018.
 */

@Data
public class DeviceInfo {
    private String fcmToken;
//    private String apiVersion;
    private DeviceType type;
    private DeviceFamily family;
    private String deviceName;

    private Map<String, String> properties;

    public Map<String, String> getProperties() {
        if(properties == null)
            properties = new HashMap<>();
        return properties;
    }
}

File: smartdevices-gateway/SmartDevicesGateway.Model/Internal/DeviceInfo.cs
//
// Copyright (c) Vogler Engineering GmbH. All rights reserved.
// Licensed under the MIT License. See LICENSE.md in the project root for license information.
//

using System.Collections.Generic;
using Newtonsoft.Json;
using SmartDevicesGateway.Model.Enums;

namespace SmartDevicesGateway.Model.Internal
{
    public class DeviceInfo
    {
        public string Id { get; set; }
        public string DeviceName { get; set; }

        public DeviceType Type { get; set; }
        public DeviceFamily family { get; set; }

        public string FcmToken { get; set; }

        //[NotMapped]
        //[JsonExtensionData]
        public Dictionary<string, object> properties { get; set; }
    }
}
